File: ru.kmorozov.api.onedrive/src/main/java/ru/kmorozov/onedrive/client/facets/HashesFacet.java
package ru.kmorozov.onedrive.client.facets;

import com.google.api.client.util.Key;

public class HashesFacet {

    @Key
    private String sha1Hash;
    @Key
    private String crc32Hash;

    public String getSha1Hash() {
        return this.sha1Hash;
    }

    public String getCrc32Hash() {
        return this.crc32Hash;
    }

    public long getCrc32() {
        // OneDrive does not always return a hash
        if (null == this.crc32Hash) {
            return 0L;
        }

        String reversed = this.crc32Hash.substring(6, 8) + this.crc32Hash.substring(4, 6) + this.crc32Hash.substring(2, 4) + this.crc32Hash.substring(0, 2);
        return Long.decode("0x" + reversed);
    }
}

File: ru.kmorozov.api.onedrive/src/main/java/ru/kmorozov/onedrive/client/facets/HashesFacet.kt
package ru.kmorozov.onedrive.client.facets

import com.google.api.client.util.Key

class HashesFacet {

    @Key
    val sha1Hash: String? = null
    @Key
    val crc32Hash: String? = null

    // OneDrive does not always return a hash
    val crc32: Long
        get() {
            if (null == crc32Hash) {
                return 0L
            }

            val reversed = crc32Hash.substring(6, 8) + crc32Hash.substring(4, 6) + crc32Hash.substring(2, 4) + crc32Hash.substring(0, 2)
            return java.lang.Long.decode("0x$reversed")
        }
}
